### PR TITLE
Utilities to convert packed dates and designations

### PR DESCRIPTION
**Pull request by moeyensj**
Comets and natural satellite support have not been added yet.

### CODE REVIEW

File: adam_core/utils/mpc.py [hunk 1]
```diff
@@ -0,0 +1,488 @@
...
+            number_pf.append(BASE62[int(x % 62)])
+            x = int(x / 62)
+
+        number_pf.reverse()
+        designation_pf = "~{}".format("".join(number_pf).zfill(4))
+
+    return designation_pf
+
+
+def _pack_provisional_designation(designation: str) -> str:
```
Comment (spenczar) [inline adam_core/utils/mpc.py:124]: Why is this private behind a `_`? I'd actually like to use it more than `pack_mpc_designation`. I don't want to accidentally pack a number.

**Review by spenczar — APPROVED** [review]
Nice!

All comments are suggestions, or just to show other ways to do things. The only one I feel strongly about at all is that `_unpack_mpc_date` should return an `astropy.time.Time`.

File: adam_core/utils/mpc.py [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+
+def _lookup_mpc(x: str) -> int:
+    # Convert the single character dates into integers.
+    try:
+        x = int(x)
+    except ValueError:
+        x = ord(x) - 55
+    if x < 0 or x > 31:
+        raise ValueError
+    return x
```
[the remark this answers is not in the document]
Comment (moeyensj) [reply adam_core/utils/mpc.py:40]: TIL.

File: adam_core/utils/mpc.py (hunk 1 above)
Comment (moeyensj) [reply adam_core/utils/mpc.py:114]: I wanted to keep it simple and let `pack_mpc_designation` take care of trying numbered, then provisional, then survey packing. Happy to make them public if it seems like they are going to be used directly.

File: adam_core/utils/mpc.py [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+    ----------
+    pf_tt : `~numpy.ndarray` (N)
+        MPC-style packed form epochs in the TT time scale.
+
+    Returns
+    -------
+    mjd_tt : `~astropy.time.core.Time` (N)
+        Epochs in TT MJDs.
+    """
+    pf_tt = np.asarray(pf_tt)
```
[the remark this answers is not in the document]
Comment (moeyensj) [reply adam_core/utils/mpc.py:50]: 'Tis. I'll remove it and trust that typing will do it.

File: adam_core/utils/mpc.py [hunk 4]
```diff
@@ -0,0 +1,488 @@
+import numpy as np
+import numpy.typing as npt
+from astropy.time import Time
+
+BASE62 = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
+BASE62_MAP = {BASE62[i]: i for i in range(len(BASE62))}
+
+
+def _unpack_mpc_date(epoch_pf: str) -> str:
```
[the remark this answers is not in the document]
Comment (moeyensj) [reply adam_core/utils/mpc.py]: There used to be this weird behavior with Time objects that if you made a Time object from a list of Time objects, it wouldn't convert the list to a single time object with multiple values and instead be a Time object of Time objects with single values. It looks like that is no longer the case though so I've done as you suggest.

**Pull request merged**